frontend/src/hooks/useDockerVisualization.ts

```diff
@@ -81,15 +87,73 @@ const useDockerVisualization = () => {
                 isVisible: true,
                 key: prev.key + 1,
             }));
-
             return currentImages;
         });
+
+        setContainers((currentContainers) => {
+            if (currentContainers.length === newContainers.length) {
+                if (isChangedContainerStatus(currentContainers, newContainers)) {
+                    setDockerOperation(DOCKER_OPERATIONS.CONTAINER_STOP);
+                    const elements = setColorToElements(newImages, newContainers);
+                    setPendingContainers(elements.initContainers);
+                    if (command.match(STATE_CHANGE_COMMAND_REGEX))
+                        setAnimation((prev) => ({
+                            isVisible: true,
+                            key: prev.key + 1,
+                        }));
+                    else setContainers(elements.initContainers);
+                }
+                return currentContainers;
+            }
+
+            setImages((currentImages) => {
+                let operation = null;
+                if (
+                    currentContainers.length < newContainers.length &&
+                    currentImages.length < newImages.length
+                )
+                    operation = DOCKER_OPERATIONS.CONTAINER_RUN;
+                else if (currentContainers.length < newContainers.length)
+                    operation = DOCKER_OPERATIONS.CONTAINER_CREATE;
+                else operation = DOCKER_OPERATIONS.CONTAINER_DELETE;
+
+                const result = setColorToElements(newImages, newContainers);
+                setPendingContainers(result.initContainers);
+                setDockerOperation(operation);
+                setAnimation((prev) => ({
+                    isVisible: true,
+                    key: prev.key + 1,
+                }));
+                return currentImages;
+            });
+
+            return currentContainers;
+        });
     };
 
-    const updateVisualizationData = async () => {
+    const isChangedContainerStatus = (
+        prevContainers: Container[],
+        currentContainers: Container[]
+    ) => {
+        const result = prevContainers.reduce<Container[]>((reducer, prevContainer) => {
+            const matchedContainer = currentContainers.find((currentContainer) => {
+                return currentContainer.id === prevContainer.id;
+            });
+            if (!matchedContainer) {
+                throw new Error('isChangedContainerStatue함수 undefind 에러');
+            }
+            if (matchedContainer.status !== prevContainer.status) {
+                reducer.push(matchedContainer);
+            }
+            return reducer;
+        }, []);
+        return result.length > 0 ? true : false;
+    };
+
```

Comment:
```suggestion
const isChangedContainerStatus = (
    prevContainers: Container[],
    currentContainers: Container[]
) => {
    return prevContainers.some(prevContainer => {
        const matchedContainer = currentContainers.find(
            currentContainer => currentContainer.id === prevContainer.id
        );
        if (!matchedContainer) {
            throw new Error('isChangedContainerStatus함수 undefined 에러');
        }
        return matchedContainer.status !== prevContainer.status;
    });
};
```

reduce 말고 some 고차함수도 괜찮아 보입니다.